ui/admin/app/abilities/credential.js

```diff
@@ -0,0 +1,14 @@
+import CredentialAbility from 'api/abilities/credential';
+import { inject as service } from '@ember/service';
+
+export default class OverrideCredentialAbility extends CredentialAbility {
+  // =service
+  @service features;
+
+  get canRead() {
+    return this.features.isEnabled('json-credentials') ||
```

Comment:
Does this make sense as the first check? Couldn't this ability be used for other credentials which would always return `true`? I would have expected us to check for `type === 'json' && features.isEnabled`

Reply:
This is being used by all credentials.

My thought behind this was if the flag is enabled then we don't care about the right side of the `||` and we carry on with the original credential ability. If the flag is disabled we then move the right side of the `||` and we check the credential type. If it is not a `json` credential we let the original credential ability do it's job. Everything else should be a `json` credential while the feature is disabled and we do not want it to be able to read.

Reply:
> My thought behind this was if the flag is enabled then we don't care about the right side of the `||`

Why is this? If for example we were in a non `json` credential _and_ the flag is on, they would all be able to read now because the boolean check will short circuit and we would never get the `super.canRead` check. Shouldn't we only be checking the flag in conjunction with the `json` type?

Reply:
Talked with @cameronperera, turns out I was misled by the formatting of this, the check short circuits against the `this.model.type !== 'json'` and not against the rest of the ternary so it works fine!